test/index.js

```diff
@@ -11,76 +11,93 @@ const rules = require('./rules.json')
 
 const TIMEOUT = 9999e3
 
-describe('DDWAF lifecycle', () => {
+describe('DDWAF', () => {
   it('should return the version', () => {
     const v = DDWAF.version()
-    assert.strictEqual([v.major, v.minor, v.patch].join('.'), pkg.libddwaf_version)
+
+    assert.strictEqual(`${v.major}.${v.minor}.${v.patch}`, pkg.libddwaf_version)
   })
 
   it('should have rulesInfo', () => {
     const waf = new DDWAF(rules)
-    assert(waf.rulesInfo)
-    assert.strictEqual(waf.rulesInfo.version, '1.3.1')
-    assert.strictEqual(waf.rulesInfo.loaded, 6)
-    assert.strictEqual(waf.rulesInfo.failed, 3)
-    assert.deepStrictEqual(waf.rulesInfo.errors, {
-      'missing key \'regex\'': [
-        'invalid_1'
-      ],
-      'invalid regular expression: *': [
-        'invalid_2',
-        'invalid_3'
-      ]
+
+    assert.deepStrictEqual(waf.rulesInfo, {
+      version: '1.3.1',
+      loaded: 6,
+      failed: 3,
+      errors: {
+        'missing key \'regex\'': [
+          'invalid_1'
+        ],
+        'invalid regular expression: *': [
+          'invalid_2',
+          'invalid_3'
+        ]
+      }
     })
   })
 
   it('should collect an attack and cleanup everything', () => {
     const waf = new DDWAF(rules)
     const context = waf.createContext()
+
     const result = context.run({
-      'server.request.headers.no_cookies': 'HELLO world',
+      'server.request.headers.no_cookies': 'value_ATTack',
```

Comment:
Are these uppercased `ATT` to test something related with case insensitive? Or it should be `value_attack`?

Reply:
Yes it's about case insensitive rules. It's not really our job to test that since it's a feature of libddwaf, but I always like to add fuziness to tests just in case.